### DISCUSSION TITLE
how do I change the "internal" name of a container?

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [louislam/dockge#599] [Ask for Help] [answered]
I am new to docker (and hence to dockge) so my terminology might be poorly chosen. Here's my best attempt to explain what I want to achieve:

I have created a [motioneye](https://github.com/motioneye-project/motioneye/wiki) container using a docker compose as shown below. The resultant motioneye webUI which is served out to my LAN has a complicated name ("35c23ae1c200") but I want it to say "motioneye" or something similar because, when I come to browse to the page, I need to remember to search for "35c23ae1c200"  or find the IP address.

```
version: "3.5"
services:
  motioneye:
    image: ghcr.io/motioneye-project/motioneye:edge
    hostname: motioneye
    environment:
      - HOSTNAME=motioneye
    ports:
      # host:container    
      #      - 8081:8081 # if streaming in UI, open this port 
      #        (or choose another in the UI and here)
      - 8765:8765
    volumes:
      # host:container    
      - /etc/localtime:/etc/localtime:ro
      - /dev/dri:/dev/dri
    restart: unless-stopped
networks: {}
```
When this runs, it  generates a container whose "internal" machine name is  `35c23ae1c200`  e.g. if  I enter the associated bash shell I find myself as  `root@35c23ae1c200`

![image](https://github.com/user-attachments/assets/6eaffc81-754b-4d69-890c-5637cca27ee4)

You can see that the machine name (presumably arbitrarily generated) is reflected in the webUI - can I change it?

### 📝 Error Message(s) or Log

dockge's motioneye log: 
motioneye-motioneye-1  | configure_logging cmd motioneye: False
motioneye-motioneye-1  | configure logging to file: None
motioneye-motioneye-1  |     INFO: hello! this is motionEye server 0.43.1b2
motioneye-motioneye-1  |     INFO: cleanup started
motioneye-motioneye-1  |     INFO: wsswitch started
motioneye-motioneye-1  |     INFO: tasks started
motioneye-motioneye-1  |     INFO: mjpg customer garbage collector has started
motioneye-motioneye-1  |     INFO: server started
motioneye-motioneye-1  |    ERROR: mjpg client timed out receiving data for camera 9 on port 8089
motioneye-motioneye-1  |  WARNING: error on read: [Errno 113] No route to host
motioneye-motioneye-1  |    ERROR: HTTP 404: Not Found (not found)
motioneye-motioneye-1  |  WARNING: 404 GET /cgi-bin/rtpd.cgi?whoami (192.168.1.115) 0.72ms
motioneye-motioneye-1  | 
motioneye-motioneye-1  | Session terminated, killing shell...    INFO: An interrupt signal received, closing …
motioneye-motioneye-1  |     INFO: server stopped
motioneye-motioneye-1  |     INFO: tasks stopped
motioneye-motioneye-1  |  ...killed.
motioneye-motioneye-1  | configure_logging cmd motioneye: False
motioneye-motioneye-1  | configure logging to file: None
motioneye-motioneye-1  |     INFO: hello! this is motionEye server 0.43.1b2
motioneye-motioneye-1  |     INFO: cleanup started
motioneye-motioneye-1  |     INFO: wsswitch started
motioneye-motioneye-1  |     INFO: tasks started
motioneye-motioneye-1  |     INFO: mjpg customer garbage collector has started
motioneye-motioneye-1  |     INFO: server started
motioneye-motioneye-1  | 
motioneye-motioneye-1  | Session terminated, killing shell...    INFO: An interrupt signal received, closing …
motioneye-motioneye-1  |     INFO: server stopped
motioneye-motioneye-1  |     INFO: tasks stopped
motioneye-motioneye-1  |  ...killed.
motioneye-motioneye-1  | configure_logging cmd motioneye: False
motioneye-motioneye-1  | configure logging to file: None
motioneye-motioneye-1  |     INFO: hello! this is motionEye server 0.43.1b2
motioneye-motioneye-1  |     INFO: cleanup started
motioneye-motioneye-1  |     INFO: wsswitch started
motioneye-motioneye-1  |     INFO: tasks started
motioneye-motioneye-1  |     INFO: mjpg customer garbage collector has started
motioneye-motioneye-1  |     INFO: server started
motioneye-motioneye-1  | 
motioneye-motioneye-1  | Session terminated, killing shell...    INFO: An interrupt signal received, closing …
motioneye-motioneye-1  |     INFO: server stopped
motioneye-motioneye-1  |     INFO: tasks stopped
motioneye-motioneye-1  |  ...killed.
motioneye-motioneye-1  | configure_logging cmd motioneye: False
motioneye-motioneye-1  | configure logging to file: None
motioneye-motioneye-1  |     INFO: hello! this is motionEye server 0.43.1b2
motioneye-motioneye-1  |     INFO: cleanup started
motioneye-motioneye-1  |     INFO: wsswitch started
motioneye-motioneye-1  |     INFO: tasks started
motioneye-motioneye-1  |     INFO: mjpg customer garbage collector has started
motioneye-motioneye-1  |     INFO: server started

### 🐻 Dockge Version

1.4.2

### 💻 Operating System and Arch

Linux Mint 22 (Ubunto 24.04)

### 🌐 Browser

chromium Version 127.0.6533.99 (Official Build) for Linux Mint (64-bit)

### 🐋 Docker Version

Docker version 24.0.7, build afdd53b

### 🟩 NodeJS Version

n/a

**Comment by trythatagain** [comment, marked as answer]
I use
 ```
  motioneye:
    container_name: name
````
Where 'name' is the container name e.i. 'motioneye', so in dockge 'motioneye' will be listed on the left panel.

**Reply by ElectronicBattle** [reply]
Crikey - it worked !  The surprise is because I thought it would be very complicated and I'd get muddled up. It wasn't; I didn't!

Thanks very much!
